pkg/apis/core/v1/defaults.go

```diff
@@ -349,6 +350,21 @@ func SetDefaults_HTTPGetAction(obj *v1.HTTPGetAction) {
 		obj.Scheme = v1.URISchemeHTTP
 	}
 }
+
+// SetDefaults_Namespace adds a default label for all namespaces
+func SetDefaults_Namespace(obj *v1.Namespace) {
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.DefaultNamespaceLabelStamping) {
+		if obj.Labels == nil {
+			obj.Labels = map[string]string{}
+		}
+		// Default label that matches the name of the namespace, for selecting w/o apriori knowledge of labels.
+		obj.Labels["kubernetes.io/metadata.name"] = obj.Name
+	} else {
+		klog.V(4).Infof("Automatic namespace stamping is disabled. not labelling %v", &obj.ObjectMeta.Name)
```

Comment:
you should check with api-machinery guys, I don't see any klog in this file,

Reply:
ack... @liggitt ok to log here?

Reply:
no, defaults get run on every read of every namespace object from etcd, avoid logging here